src/NuGet.Clients/PackageManagement.UI/Feeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -147,7 +132,14 @@ internal class MultiSourcePackageMetadataProvider : IPackageMetadataProvider
             var completed = (await Task.WhenAll(tasks))
                 .Where(m => m != null);
 
-            return completed.SelectMany(p => p);
+            var packages = completed.SelectMany(p => p);
+
+            var uniquePackages = packages
+                .GroupBy(
+                    m => m.Identity.Version,
+                    (v, ms) => ms.First());
```

Comment:
`packages` may contain multiple packages coming from different sources with the same identity.